torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -1256,12 +1376,64 @@ def _are_inputs_layout_compatible(self, layouts: List[Layout]) -> bool:
         K = max(A_size[-1], B_size[-2])
         M = A_size[-2]
         N = B_size[-1]
-        if K != A_size[-1]:
+        if K != A_size[-1] and K != 2 * A_size[-2]:
```

Comment:
Why does K have to equal twice the size of the first dimension of A on this line?